compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1073,6 +1074,16 @@ rustc_queries! {
         cache_on_disk_if { true }
     }
 
+    /// Evaluate a static's initializer, returning the allocation of the initializer's memory.
+    query eval_static_initializer(key: DefId) -> EvalStaticInitializerRawResult<'tcx> {
+        desc { |tcx|
+            "evaluating initializer of static `{}`",
+            tcx.def_path_str(key)
+        }
+        cache_on_disk_if { true }
```

Comment:
```suggestion
        cache_on_disk_if { key.is_local() }
```
For non-local keys, they are loaded from metadata, so no need to cache them a second time.